Mathlib/Tactic/Simps/Basic.lean

```diff
@@ -1185,6 +1204,15 @@ def simpsTac (ref : Syntax) (nm : Name) (cfg : Config := {})
   let lhs : Expr := mkConst d.name <| d.levelParams.map Level.param
   let todo := todo.eraseDups |>.map fun (proj, stx) ↦ (proj ++ "_", stx)
   let mut cfg := cfg
+  let nm : NameStruct :=
+    { parent := nm.getPrefix
+      components :=
+        if let .some n := cfg.nameStem then
+          if n == "" then [] else [n]
+        else
+          let s := nm.lastComponentAsString
+          -- TODO: how can we tell if this is an instance? The `instance` attribute hasn't run yet.
```

Comment:
Oh, that is a bit unfortunate. Can we run `simps` at a later stage (after compilation?) where the `instance` attribute has run?

Reply:
Maybe @Kha knows? My guess is that `@[attr] instance foo` becomes `@[attr, instance] def foo`, which is why things currently behave as they do.